### PR TITLE
RD-4376 Blueprint tile improvement

### PR DESCRIPTION
**Pull request by leon-good-life**
Jira ticket: https://cloudifysource.atlassian.net/browse/RD-4376

## Description
This PR removes all fields from blueprint tiles except the following:
- Description - if the description is more than 100 characters it should be trimmed as shown in the screenshot
- Creator
- State
- Deployments

This PR should also Minimize the width as well so it will be compact, fit 5-6 items in a row vs currently 3

## Screenshots / Videos
Screen Shot 2022-03-06 at 13 47 46

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
All tests are passing except 2 in `getting_started_spec` that failing also in `master` branch.

System tests after I updated `user_spec`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2185/pipeline

I started system tests: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2184/pipeline

 

## Documentation
The Blueprints Catalog page is already updated: https://docs.cloudify.co/latest/working_with/console/widgets/blueprintcatalog/

### CODE REVIEW

**Comment by leon-good-life** [comment]
`user_spec` is failing and needs to be updated.

**Review by qooban — COMMENTED** [review]
RD-4376 was not refined yet 🙂 Maybe the idea is quite simple, but IMHO we shouldn't remove those fields completely from the view, we should add a field in widget configuration allowing user to specify which parameters to show.

Please search for `fieldsToShow` configuration parameter, it is used in many widgets. 
Check solution from the Blueprints Catalog widget.

**Review by qooban — COMMENTED** [review]
Added some comments.

Tested quickly the UI. Take a look at:


https://user-images.githubusercontent.com/5202105/158324176-b9052a54-2339-47b0-b8ba-6ba8ed6c61ae.mp4

I think that for smaller screens it doesn't look good. Can you make blueprint name to be displayed properly in smaller screens?

**Comment by leon-good-life** [comment]
I fixed the title in small screens

File: widgets/blueprints/src/widget.tsx [hunk 1]
```diff
@@ -23,6 +23,14 @@ Stage.defineWidget<unknown, unknown, BlueprintsWidgetConfiguration>({
     initialConfiguration: [
         Stage.GenericConfig.POLLING_TIME_CONFIG(10),
         Stage.GenericConfig.PAGE_SIZE_CONFIG(5),
+        {
+            id: 'fieldsToShow',
+            name: 'List of fields to show in the blueprint tile',
+            placeHolder: 'Select fields from the list',
+            items: ['Creator', 'State', 'Deployments'],
+            default: 'Creator,State,Deployments',
+            type: Stage.Basic.GenericField.MULTI_SELECT_LIST_TYPE
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/widget.tsx]: Added translations: 177eaa9

File: widgets/blueprints/src/widget.tsx [hunk 2]
```diff
@@ -23,6 +23,14 @@ Stage.defineWidget<unknown, unknown, BlueprintsWidgetConfiguration>({
     initialConfiguration: [
         Stage.GenericConfig.POLLING_TIME_CONFIG(10),
         Stage.GenericConfig.PAGE_SIZE_CONFIG(5),
+        {
+            id: 'fieldsToShow',
+            name: 'List of fields to show in the blueprint tile',
+            placeHolder: 'Select fields from the list',
+            items: ['Creator', 'State', 'Deployments'],
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/widget.tsx]: I added `Created` and `Updated`: [3357e6e](https://github.com/cloudify-cosmo/cloudify-stage/pull/1863/commits/3357e6edbfc0b8b6bed5ce5907a1b38683c0c59b)

File: widgets/blueprints/src/BlueprintsCatalog.tsx [hunk 3]
```diff
@@ -47,67 +48,79 @@ export default function BlueprintsCatalog({
...
 
-                        <Grid.Row className="noPadded">
-                            <Grid.Column width="7">
-                                <Header as="h5" icon textAlign="left">
-                                    Updated
-                                </Header>
-                            </Grid.Column>
-                            <Grid.Column width="9">{item.updated_at}</Grid.Column>
-                        </Grid.Row>
+                        {fieldsToShow.includes('Created') && (
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/BlueprintsCatalog.tsx:77]: There is no `DataTable` component in `BlueprintsCatalog.tsx` file, the tiles implemented using `<Grid>` And `<Grid.Row  show={false}>` (just for check) doesn't work. Do you mean to implement the tiles using `DataTable` component.

[the remark this answers is not in the document]
Comment (qooban) [reply widgets/blueprints/src/BlueprintsCatalog.tsx:77]: > Do you mean to implement the tiles using DataTable component.

No, I'm saying that `fieldsToShow` configuration field should affect both views - `Table` and `Catalog`. That means we should have also show/hide mechanism when Blueprints widget is in Table view. This comment is actually about doing a change in `BlueprintsTable`, not in `BlueprintsCatalog`.

[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/BlueprintsCatalog.tsx:77]: I updated `BlueprintsTable`: [260eff4](https://github.com/cloudify-cosmo/cloudify-stage/pull/1863/commits/260eff4b7b1d21eae80db4620b8295a4256b41cc)

File: widgets/blueprints/src/widget.tsx [hunk 4]
```diff
@@ -23,6 +23,14 @@ Stage.defineWidget<unknown, unknown, BlueprintsWidgetConfiguration>({
...
+            default: 'Creator,State,Deployments',
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/widget.tsx:31]: I now limit in `blueprints.json`: [260eff4](https://github.com/cloudify-cosmo/cloudify-stage/pull/1863/commits/260eff4b7b1d21eae80db4620b8295a4256b41cc)

**Comment by leon-good-life** [comment]
I started system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2243/pipeline

**Review by qooban — COMMENTED** [review]
Looks better. Thanks for addressing some of the comments.

I've discovered that with 5 tiles the full width of the widget is not used, see screenshot:

image

Do you think we can fix it?

File: widgets/blueprints/src/widget.tsx [hunk 5]
```diff
@@ -23,6 +23,14 @@ Stage.defineWidget<unknown, unknown, BlueprintsWidgetConfiguration>({
     initialConfiguration: [
         Stage.GenericConfig.POLLING_TIME_CONFIG(10),
         Stage.GenericConfig.PAGE_SIZE_CONFIG(5),
+        {
+            id: 'fieldsToShow',
+            name: tCatalogConfiguration('fieldsToShow.label'),
+            placeHolder: tCatalogConfiguration('pfieldsToShow.laceholder'),
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/widget.tsx]: fixed: b31dd992e44daa3c5cb63a373b0f324747058848

[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/widget.tsx]: Added: [cbe3d34](https://github.com/cloudify-cosmo/cloudify-stage/pull/1863/commits/cbe3d34adc34feff9a46451f6d58084c3fde497c)

File: widgets/blueprints/src/BlueprintsTable.tsx [hunk 6]
```diff
@@ -15,6 +15,8 @@ export default function BlueprintsTable({
     const { DataTable, Icon, Image, ResourceVisibility } = Stage.Basic;
     const { BlueprintActions } = Stage.Common;
     const tableName = 'blueprintsTable';
+    const { fieldsToShow } = widget.configuration;
+    console.log(fieldsToShow);
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/BlueprintsTable.tsx]: removed: [35d50b7](https://github.com/cloudify-cosmo/cloudify-stage/pull/1863/commits/35d50b78449fb0a23d0ee2ebbd726047c6e6cd42)

File: widgets/blueprints/src/widget.tsx [hunk 7]
```diff
@@ -23,6 +23,18 @@ Stage.defineWidget<unknown, unknown, BlueprintsWidgetConfiguration>({
...
+        {
+            id: 'fieldsToShow',
+            name: t('configuration.fieldsToShow.label'),
+            placeHolder: t('configuration.fieldsToShow.placeholder'),
+            items: ['Created', 'Updated', 'Creator', 'State', 'Deployments'].map(item =>
+                t(`configuration.fieldsToShow.items.${item}`)
+            ),
+            default: join(
+                ['Creator', 'State', 'Deployments'].map(item => t(`configuration.fieldsToShow.items.${item}`))
+            ),
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply widgets/blueprints/src/widget.tsx]: eb4a608

**Comment by leon-good-life** [comment]
1. Fix the bottom padding after the deploy button.
2. Fix last row.
3. horizontal scroll.

**Comment by leon-good-life** [comment]
I started system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2259/pipeline

File: widgets/blueprints/src/widget.tsx [hunk 8]
```diff
@@ -23,6 +23,14 @@ Stage.defineWidget<unknown, unknown, BlueprintsWidgetConfiguration>({
     initialConfiguration: [
```
[the remark this answers is not in the document]
Comment (qooban) [reply widgets/blueprints/src/widget.tsx:25]: I guess you are not planning to migrate this file?

**Review by qooban — APPROVED** [review]

**Comment by leon-good-life** [comment]
Maybe I will migrate [widgets/blueprints/src/widget.tsx](https://github.com/cloudify-cosmo/cloudify-stage/pull/1863/files/a9015c80739d2636e145146b01547a1ff0463256#diff-f6c03f60ad37aab53b8f883f74cb38432ad6df96500871ce600aee3e725f56a2) in the future

**Comment by leon-good-life** [comment]
blueprint_catalog_spec is passing in this branch:
Screen Shot 2022-03-21 at 13 33 26

**Review by qooban — APPROVED** [review]

**Pull request merged**